test/python/test_docker_pack.py

```diff
@@ -0,0 +1,288 @@
+        if 'entering the event loop' in container_logs:
+            return True
+
+        time.sleep(1)
+
+    return False
+
+
+# ########
+# Fixtures
+# ########
+@pytest.fixture(scope="module")
+def docker_client():
+    client = docker.from_env()
+    return client
+
+
+@pytest.fixture(scope="module")
+def docker_image(module_tmpdir, project_path, prepare_ignore, request, docker_client):
+    cmd = [os.path.join(basepath, "cartridge"), "pack", "docker", project_path]
+    process = subprocess.run(cmd, cwd=module_tmpdir)
+    assert process.returncode == 0, \
+        "Error during creating of docker image"
+
+    image_name = find_image(docker_client, project_name)
+    assert image_name is not None, "Docker image isn't found"
+
+    def delete_image(image_name):
+        if docker_client.images.list('myapp:0.1.0-0'):
+            # remove all image containers
+            containers = docker_client.containers.list(
+                all=True,
+                filters={'ancestor': image_name}
+            )
+
+            for c in containers:
+                c.remove(force=True)
+
+            # remove image itself
+            docker_client.images.remove(image_name)
+
+    request.addfinalizer(lambda: delete_image(image_name))
+    return {'name': image_name}
+
+
+# #####
+# Tests
+# #####
+def test_invalid_base_dockerfile(project_path, module_tmpdir, tmpdir):
+    invalid_dockerfile_path = os.path.join(tmpdir, 'Dockerfile')
+    with open(invalid_dockerfile_path, 'w') as f:
+        f.write('''
+            # Invalid dockerfile
+            FROM ubuntu:xenial
+        ''')
+
+    cmd = [
+        os.path.join(basepath, "cartridge"),
+        "pack", "docker",
+        "--from", invalid_dockerfile_path,
+        project_path,
+    ]
+    process = subprocess.run(cmd, cwd=module_tmpdir)
+    assert process.returncode == 1
+
+
+def test_docker_pack(project_path, docker_image, tmpdir, docker_client):
+    image_name = docker_image['name']
+    container = docker_client.containers.create(image_name)
+
+    container_distribution_dir = '/usr/share/tarantool/{}'.format(project_name)
+
+    # check if distribution dir was created
+    command = '[ -d "{}" ] && echo true || echo false'.format(container_distribution_dir)
+    output = run_command_on_image(docker_client, image_name, command)
+    assert output == 'true'
+
+    # get distribution dir contents
+    arhive_path = os.path.join(tmpdir, 'distribution_dir.tar')
+    with open(arhive_path, 'wb') as f:
+        bits, _ = container.get_archive(container_distribution_dir)
+        for chunk in bits:
+            f.write(chunk)
+
+    with tarfile.open(arhive_path) as arch:
+        arch.extractall(path=os.path.join(tmpdir, 'usr/share/tarantool'))
+    os.remove(arhive_path)
+
+    assert_dir_contents(
+        recursive_listdir(os.path.join(tmpdir, 'usr/share/tarantool/', project_name)),
+        skip_tarantool_binaries=True
+    )
+
+    assert_filemodes(tmpdir)
+    container.remove()
```

Comment:
We don't destroy container if tests failes because of assert_filemodes. Consider adding of finalizer and moving a bit higher

Reply:
`docker_image` finalizer removes all containers based on this image.